ui/ui-components/pages/model/[modelId]/destination/[destinationId]/data.tsx

```diff
@@ -858,41 +850,38 @@ export default function Page(props) {
 Page.getInitialProps = async (ctx: NextPageContext) => {
   const client = generateClient(ctx);
   const { destinationId, modelId } = ctx.query;
-  const { destination } = await client.request(
+  const { destination } = await client.request<Actions.DestinationView>(
     "get",
     `/destination/${destinationId}`
   );
   ensureMatchingModel("Destination", destination.modelId, modelId.toString());
-  const { groups } = await client.request("get", `/groups`, {
-    modelId: destination?.modelId,
-  });
-  const { properties } = await client.request("get", `/properties`, {
-    state: "ready",
-    modelId: destination?.modelId,
-  });
-
-  let mappingOptions = {};
-  let destinationTypeConversions = {};
-  let exportArrayProperties = [];
-  let hydrationError: Error;
+  const { groups } = await client.request<Actions.GroupsList>(
+    "get",
+    `/groups`,
+    {
+      modelId: destination?.modelId,
+    }
+  );
+  const { properties } = await client.request<Actions.PropertiesList>(
+    "get",
+    `/properties`,
+    {
+      state: "ready",
+      modelId: destination?.modelId,
+    }
+  );
 
-  try {
-    const mappingOptionsResponse = await client.request(
+  const { options: mappingOptions, destinationTypeConversions } =
+    await client.request<Actions.DestinationMappingOptions>(
       "get",
       `/destination/${destinationId}/mappingOptions`
     );
-    mappingOptions = mappingOptionsResponse.options;
-    destinationTypeConversions =
-      mappingOptionsResponse.destinationTypeConversions;
 
-    const exportArrayPropertiesResponse = await client.request(
+  const { exportArrayProperties } =
+    await client.request<Actions.DestinationExportArrayProperties>(
       "get",
       `/destination/${destinationId}/exportArrayProperties`
     );
-    exportArrayProperties = exportArrayPropertiesResponse.exportArrayProperties;
-  } catch (error) {
-    hydrationError = error.toString();
```

Comment:
This custom hydrationError was causing a lot of trouble so now it just falls back to the hydration error set by the next app.